docarray/array/mixins/find.py

```diff
@@ -158,3 +160,22 @@ def _find(
         self, query: 'ArrayType', limit: int, **kwargs
     ) -> Tuple['np.ndarray', 'np.ndarray']:
         raise NotImplementedError
+
+    def _filter(
+        self,
+        query: Dict,
+        limit: Optional[Union[int, float]] = None,
```

Comment:
in jina flow, the `limit` sometimes comes from `parameters` (e.g,. TagStructureView). Then, all of the integers in `parameters` will be enforced cast to float. But I agree that it is kinda weird here to use limit as float, I think it can be removed.

Reply:
oh, I'm worried that this would break the executors in hub. I will keep it.